translation-stackdriver/src/main/java/zipkin2/translation/stackdriver/SpanTranslator.java

```diff
@@ -71,43 +82,22 @@
         spanBuilder.setEndTime(endTime);
       }
     }
-    spanBuilder.putAllLabels(labelExtractor.extract(zipkinSpan));
-    if (logTranslation) LOG.log(FINE, "<< translated to stackdriver span: {0}", spanBuilder);
-    return spanBuilder;
-  }
+    spanBuilder.setAttributes(ATTRIBUTES_EXTRACTOR.extract(zipkinSpan));
 
-  private static long parseUnsignedLong(String lowerHex) {
-    if (lowerHex == null) return 0;
-    long result = 0;
-    for (int i = 0; i < 16; i++) {
-      char c = lowerHex.charAt(i);
-      result <<= 4;
-      if (c >= '0' && c <= '9') {
-        result |= c - '0';
-      } else if (c >= 'a' && c <= 'f') {
-        result |= c - 'a' + 10;
-      } else {
-        return 0;
+    if (!zipkinSpan.annotations().isEmpty()) {
+      TimeEvents.Builder events = TimeEvents.newBuilder();
+      for (Annotation annotation : zipkinSpan.annotations()) {
+        events.addTimeEvent(TimeEvent.newBuilder()
+            .setTime(createTimestamp(annotation.timestamp()))
+            .setAnnotation(TimeEvent.Annotation.newBuilder()
+                .setDescription(toTruncatableStringProto(annotation.value())))
+        );
       }
+      spanBuilder.setTimeEvents(events);
     }
-    return result;
-  }
 
-  private static SpanKind getSpanKind(/* Nullable */ Span.Kind zipkinKind) {
-    // Stackdriver Trace still does not have any match for CONSUMER or PRODUCER, and sending it as
-    // UNRECOGNIZED triggers an error.
-    if (zipkinKind == null
-        || zipkinKind == Span.Kind.CONSUMER
-        || zipkinKind == Span.Kind.PRODUCER) {
-      return SpanKind.SPAN_KIND_UNSPECIFIED;
-    }
-    if (zipkinKind == Span.Kind.CLIENT) {
-      return SpanKind.RPC_CLIENT;
-    }
-    if (zipkinKind == Span.Kind.SERVER) {
-      return SpanKind.RPC_SERVER;
-    }
-    return SpanKind.UNRECOGNIZED;
+    if (logTranslation) LOG.log(FINE, "<< translated to stackdriver span: {0}", spanBuilder);
```

Comment:
kind should end up somewhere at least making fake annotations.

https://github.com/openzipkin/zipkin/blob/master/zipkin-ui/js/component_ui/spanRow.js#L113